File: js/topic.js
// create topic

const Kafka = require("kafkajs").Kafka;
const getIpAdress = require("./host_ip.js");
const ipAddress = getIpAdress();
const brokerPort = "9092"

run();

async function run(){
    try {
        const kafka = new Kafka({
            "clientId": "admin",
            "brokers": [ipAddress.concat(":").concat(brokerPort)]
        })
        const admin = kafka.admin();
        console.log("Connecting...")
        await admin.connect()
        console.log("Connected!")

        // Partition by A-M and N-Z
        await admin.createTopics({
            "topics": [{
                "topic": "Users",
                "numPartitions": 2
            }]
        })
        console.log("Topic created.")
        await admin.disconnect()
    }
    catch(ex) {
        console.log("Error occurred.")
    }
    finally{
        process.exit(0)
    }
}

File: python/topic.py
import kafka
import logging

logging.basicConfig(format='%(levelname)s: %(message)s', level=logging.INFO)
brokerPort = "9092"

try:
    admin = kafka.KafkaAdminClient(
        bootstrap_servers="127.0.0.1:" + brokerPort, 
        client_id="admin"
    )
    logging.info("Connect successfully!")

    # create topic
    topics = []
    topics.append(kafka.admin.NewTopic(
        name = "Users",
        num_partitions=2,
        replication_factor=1
    ))

    admin.create_topics(topics)
    logging.info(f"Topic created.")

    # delete topic
    # admin.delete_topics(topics=["Users"])
    # logging.info("Topic deleted.")
except Exception as e:
    logging.error(e)